app/lib/proof_provider/keybase.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class ProofProvider::Keybase
+  BASE_URL = 'https://keybase.io'
+
+  class Error < StandardError; end
+
+  class ExpectedProofLiveError < Error; end
+
+  class UnexpectedResponseError < Error; end
+
+  def initialize(proof = nil)
+    @proof = proof
+  end
+
+  def serializer_class
+    ProofProvider::Keybase::Serializer
+  end
+
+  def worker_class
+    ProofProvider::Keybase::Worker
+  end
+
+  def validate!
+    unless @proof.token&.size == 66
+      @proof.errors.add(:base, I18n.t('identity_proofs.errors.keybase.invalid_token'))
```

Comment:
It feels really strange to reach into `@proof` to modify it's errorset. Why not keep errors here?

Reply:
`AccountIdentityProof` calls this method from `validate`. It is essentially an ActiveRecord validator. I was thinking whether it should be its own class or not, but it's not too many lines, so I just put it here.

One problem is that I have to use `validate` with a method/lambda that calls `provider_instance`, instead of `validates_with SomeValidatorClass` because that class would depend on the dynamic `provider` value. Of course I could do something like `validates_with ProofProvider::Validator`, which then in its own `validate_each` method would resolve the right class and call it, but why do all those extra steps when this method works?

Reply:
I guess. This violates the Law of Demeter, though, and IMO will create annoyances down the line.